tests/permissive_mode/permissive_mode.c

```diff
@@ -26,3 +28,24 @@ Test(permissive_mode, main) {
 
     cr_assert(ia2_get_pkru() == 0xFFFFFFF0);
 }
+
+__thread int tls;
+
+void *start_thread(void *const arg) {
+  // Mostly bogus stuff.
+  // Just want to use stack, heap, and TLS.
+  int a[4096] = {0};
+  malloc(10);
+  return (void *)((void *)a - (void *)&tls);
+}
+
+Test(permissive_mode, multithreaded) {
+  pthread_t threads[10];
+  for (size_t i = 0; i < 10; i++) {
+#if IA2_ENABLE
+    pthread_create(&threads[i], NULL, start_thread, NULL);
+#endif
+  }
+  // Exit before joining threads.
```

Comment:
Is there a particular reason not to join threads?

Reply:
Presumably because if we were cleaning up thread stacks properly then joining everything means we wouldn't print any thread stacks or TLS at the end.

Reply:
Yeah, exactly.  I wanted to see what the `/proc/self/maps` state is with all of the threads still existing.

Reply:
Makes sense, it would be nice if the comment also explained that intent.

Reply:
Fixed in 3f9ef14c9.